drivers/reset/Kconfig

```diff
@@ -0,0 +1,32 @@
+# Reset Controller driver configuration options
+
+# Copyright (c) 2022 Andrei-Edward Popa <andrei.popa105@yahoo.com>
+# SPDX-License-Identifier: Apache-2.0
+
+#
+# Reset Controller options
+#
+menuconfig RESET
+	bool "Reset Controller drivers"
+	help
+	  Reset Controller drivers. Reset controller node represents
+	  a region containing the base address of a reset controller device.
+	  The typical use-case is for some other node's drivers to acquire a
+	  reference to the reset controller node together with some reset
+	  information such as offset to reset register and reset bit and
+	  use syscalls to call the reset controller API functions.
```

Comment:
Indeed what you are describing here is a very special kind of reset controller. I understand that this probably is  99% of the reset controllers out there, but I think we should be more generic.